### PR TITLE
Add authorization Rules

### PR DESCRIPTION
**Pull request by momo-87**
### Added Features

- [x] Install CanCanCan in the project
- [x] Implement the post-deleting functionality
- [x] Implement the comment-deleting functionality

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def new
   def show
     @user = User.find_by(id: params[:user_id])
     @post = @user.posts.find(params[:id])
-    @current_user = current_user
+    @currrent = current_user
     @like = Like.new
   end
```
Comment (Qoosim) [inline app/controllers/posts_controller.rb:21]: - [ ] It is required to use cancancan to add  post authorization to your post controller. This should be added on the first line of your controller. Something like this. :point_down: 

```
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (momo-87) [reply app/controllers/posts_controller.rb:21]: @Qoosim.
Thank you for reviewing my code. I really appreciate the time and effort you invested, but I guess the requirement is not to use CanCanCan to add post authorization but to set the delete post authorization when implementing the deleting post functionality (see image below):
image

Below are the authorization I set in the Ability class:
image

**Review by Nemwel-Boniface — COMMENTED** [review]
## Hello @momo-87  👋,

I am Nemwel your reviewer.


## _This is an additional review as a result of the second opinion request._

I have received your second opinion request with a complaint about the previous reviewer @Qoosim  Below are my findings:

I have found that the requested change by the reviewer to be correct. Basically the ```load_and_authorize_resource```  is a key part of CancanCan that simplifies the process of loading and authorizing resources in your controllers. By including this line, you are able to simplify the process of:
1. **Resource loading** - In your project, you have various types of resources (e.g., blogs, users, comments) that you want to manage. To perform actions like viewing, editing, or deleting these resources, you need to first find and load the relevant data from your database.
2. **Authorization** - After loading a resource, you often want to control who is allowed to perform certain actions on that resource. For instance, you might want to make sure that only the owner of an blog can edit it, or only an admin user can delete comments.

And much more, the above two are just an example of what you can do.


**What you should do:**

- [ ] As the change requested is actually valid, kindly ensure that you make the change that was requested by the last reviewer and when you are done please feel free to request another review in your dashboard. 

**Bonus:** To better help you to understand more on what the line does, I gathered a few resources for you. [This](https://stackoverflow.com/questions/30840231/cancan-explanation-of-load-and-authorize-resource) [this](https://itecnote.com/tecnote/ruby-on-rails-how-to-understand-load_resource-and-authorize_resource-methods-of-cancancan-gem/) [The official docs](https://github.com/seven1m/load_and_authorize_resource).


In case anything is not 100% clear tag me @Nemwel-Boniface so that I get the notification.

Keep rocking 🚀!

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi Team,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**